contrib/olivere/elastic/elastictrace_test.go

```diff
@@ -197,3 +202,72 @@ func TestQuantize(t *testing.T) {
 		assert.Equal(t, tc.expected, quantize(tc.url, tc.method))
 	}
 }
+
+func TestPeek(t *testing.T) {
+	assert := assert.New(t)
+
+	for _, tt := range [...]struct {
+		max  int           // content length
+		rc   io.ReadCloser // stream
+		n    int           // bytes to peek at
+		snip string        // expected snippet
+		err  error         // expected error
+	}{
+		0: {
+			// extract 3 bytes from a content of length 7
+			rc:   ioutil.NopCloser(bytes.NewBufferString("ABCDEFG")),
+			max:  7,
+			n:    3,
+			snip: "ABC",
+		},
+		1: {
+			// extract 7 bytes from a content of length 7
+			rc:   ioutil.NopCloser(bytes.NewBufferString("ABCDEFG")),
+			max:  7,
+			n:    7,
+			snip: "ABCDEFG",
+		},
+		2: {
+			// extract 100 bytes from a content of length 9 (impossible scenario)
+			rc:   ioutil.NopCloser(bytes.NewBufferString("ABCDEFG")),
+			max:  9,
+			n:    100,
+			snip: "ABCDEFG",
+		},
+		3: {
+			// extract 5 bytes from a content of length 2 (impossible scenario)
+			rc:   ioutil.NopCloser(bytes.NewBufferString("ABCDEFG")),
+			max:  2,
+			n:    5,
+			snip: "AB",
+		},
+		4: {
+			rc:  ioutil.NopCloser(bytes.NewBufferString("ABCDEFG")),
+			max: 0,
+			err: errors.New("empty stream"),
+		},
+		5: {
+			n:   4,
+			max: 4,
+			err: errors.New("empty stream"),
+		},
+		6: {
+			rc:   ioutil.NopCloser(bytes.NewBufferString("ABCDEFG")),
+			n:    4,
+			max:  -1,
+			snip: "ABCD",
+		},
+	} {
+		snip, rc, err := peek(tt.rc, tt.max, tt.n)
+		assert.Equal(tt.err, err)
+		assert.Equal(tt.snip, snip)
+
+		if tt.rc != nil {
+			// if a non-nil io.ReadCloser was sent, the returned io.ReadCloser
+			// must always return the entire original content.
+			all, err := ioutil.ReadAll(rc)
+			assert.Nil(err)
+			assert.Equal("ABCDEFG", string(all))
```

Comment:
`"ABCDEFG"` could be a variable but it's a nitpick

Reply:
No, you're right. I will change this.